docs/guides/i18n.rst

```diff
@@ -36,6 +38,24 @@ ordered later on the Admin > Plugins page:
 
     Unless you are overriding core's or another plugin's language strings, it is good practice for the language keys to start with your plugin name. For example: ``yourplugin:success``, ``yourplugin:title``, etc. This helps avoid conflicts with other language keys.
 
+.. warning:: Do not use spaces in your language keys. The JS API does not support them fully.
+
+Optimized client-side loading
+-----------------------------
+
+Elgg plans to optimize client-side language loading in the future. Plugins may prepare for this by providing a list of language keys that are likely to be needed (via ``elgg/echo``) on the home page or other common URLs.
+
+.. code-block:: php
+
+    <?php // in mod/example/elgg-plugin.php
+
+    return [
+        'home_language_keys' => [
```

Comment:
I guess I figured the home page was the likeliest entry point. Better ideas? `js_language_keys`?